vertical-pod-autoscaler/pkg/utils/metrics/admission/admission.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package admission (aka metrics_admission) - code for metrics of VPA Admission Controller plugin
+package admission
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/metrics"
+)
+
+const (
+	metricsNamespace = metrics.TopMetricsNamespace + "admission_controller"
+)
+
+// AdmissionLatency measures latency / execution time of Admission Control execution
+// usual usage pattern is: timer := AdmissionLatency(...) ; compute ; timer.Observe()
+type AdmissionLatency struct {
+	histo *prometheus.HistogramVec
+	start time.Time
+}
+
+// AdmissionStatus describes the result of Admission Control execution
+type AdmissionStatus string
+
+// AdmissionResource describes the resource processed by Admission Control execution
+type AdmissionResource string
+
+const (
+	// Error denotes a failed Admission Control execution
+	Error AdmissionStatus = "error"
+	// Skipped denotes an Admission Control execution w/o applying a recommendation
+	Skipped AdmissionStatus = "skipped"
+	// Applied denotes an Admission Control execution when a recommendation was applied
+	Applied AdmissionStatus = "applied"
+)
+
+const (
+	// Unknown means that the resource was not yet determined
+	Unknown AdmissionResource = "n/a"
```

Comment:
I'd go for "unknown". And I don't agree with the comment, it seems from it that the resource will be determined in the future, but this is rather - we were not able to determine the resource (e.g. unmarshalling error)

Reply:
Thanks, fixed.